src/token/infoerror-token-parser.ts

```diff
@@ -12,43 +14,59 @@ interface TokenData {
   lineNumber: number;
 }
 
-function parseToken(parser: Parser, options: ParserOptions, callback: (data: TokenData) => void) {
-  // length
-  parser.readUInt16LE(() => {
-    parser.readUInt32LE((number) => {
-      parser.readUInt8((state) => {
-        parser.readUInt8((clazz) => {
-          parser.readUsVarChar((message) => {
-            parser.readBVarChar((serverName) => {
-              parser.readBVarChar((procName) => {
-                (options.tdsVersion < '7_2' ? parser.readUInt16LE : parser.readUInt32LE).call(parser, (lineNumber: number) => {
-                  callback({
-                    'number': number,
-                    'state': state,
-                    'class': clazz,
-                    'message': message,
-                    'serverName': serverName,
-                    'procName': procName,
-                    'lineNumber': lineNumber
-                  });
-                });
-              });
-            });
-          });
-        });
-      });
-    });
-  });
+function readToken(buf: Buffer | BufferList, offset: number, options: ParserOptions): Result<TokenData> {
+  let tokenLength;
+  ({ offset, value: tokenLength } = readUInt16LE(buf, offset));
+
```

Comment:
is the token length that we got here represent the size for entire info error token? 
if true, if we compare the buf.length with tokenLength +offset, will this always fail the if statement? 
under the connection retry test, we have a mock token size of 31 byte, and here when we call to readtoken, the offset is 3, the buffer length is 31, the logic never pass through here with the pervious compare if(bur.length()< tokenlength +offset)
I change it to (buf.length()+offset<tokenLength) means that if incoming buffer size plus the current start offset is less then the tokenlength that we expect, then we throw and wait for more. 🤔